src/libAtomVM/nifs.c

```diff
@@ -1284,14 +1284,42 @@ static term nif_erlang_spawn(Context *ctx, int argc, term argv[])
 static term nif_erlang_send_2(Context *ctx, int argc, term argv[])
 {
     UNUSED(argc);
+    term target = argv[0];
+    GlobalContext *glb = ctx->global;
+    struct ListHead *processes_table_list = synclist_rdlock(&glb->processes_table);
+
+    if (term_is_pid(target)) {
+        int32_t target_pid = term_to_local_process_id(target);
+
+        globalcontext_send_message_nolock(glb, target_pid, argv[1]);
+        synclist_unlock(&glb->processes_table);
 
-    term pid_term = argv[0];
-    VALIDATE_VALUE(pid_term, term_is_pid);
+    } else if (term_is_atom(target)) {
+        Context *p = NULL;
+        int atom_index = term_to_atom_index(target);
 
-    int local_process_id = term_to_local_process_id(pid_term);
-    globalcontext_send_message(ctx->global, local_process_id, argv[1]);
+        int test_pid = globalcontext_get_registered_process(glb, atom_index);
+        if (UNLIKELY(test_pid == 0)) {
+            goto badarg_error;
+        }
+
+        p = globalcontext_get_process_nolock(glb, test_pid);
+        if (UNLIKELY(IS_NULL_PTR(p))) {
+            goto badarg_error;
+        } else {
+            int32_t target_pid = p->process_id;
```

Comment:
I suggest `target_id` or `target_local_id`.

Reply:
I have no objections to changing the variable name here, I just chose `target_pid` for consistency, as it is used in all of the other nifs.c functions that operate on, or return process ids.

Reply:
I think I have used more meaningful variable names now, and I moved the declaration of receiver_id to the top since it is used in both branches.